test/edu-benefits/utils/veteran.unit.spec.js

```diff
@@ -41,6 +41,49 @@ describe('veteranToApplication', () => {
 
     expect(applicationData.toursOfDuty[0].fromDate).to.equal('1996-09-06');
   });
+  it('converts date ranges to iso', () => {
+    const formData = createVeteran();
+    formData.toursOfDuty.push({
+      serviceBranch: makeField('army'),
+      dateRange: {
+        from: {
+          month: makeField('9'),
+          day: makeField('6'),
+          year: makeField('1996')
+        },
+        to: {
+          month: makeField('9'),
+          day: makeField('6'),
+          year: makeField('1997')
+        }
+      }
+    });
+    const applicationData = JSON.parse(veteranToApplication(formData));
+
+    expect(applicationData.toursOfDuty[0].dateRange.from).to.equal('1996-09-06');
+    expect(applicationData.toursOfDuty[0].dateRange.to).to.equal('1997-09-06');
+  });
+  it('removes partial date ranges', () => {
```

Comment:
I struggled with this a bit when I was working on the graduation date ticket (where the user only needs to enter a month and a year).

In order for something to be a `date`, it technically needs precision down to the millisecond.  If the input isn't precise, Javascript assumes "the first possible moment that this could be":

```
new Date('December 2014')
Mon Dec 01 2014 00:00:00 GMT-1000 (HST)
``` 

This leads to problems when trying to check the difference between dates, since the user didn't necessarily mean "the first of the month".  The right thing to do here is probably to create a new type of object for partial dates, or to rework the way our validations happen.